File: client-ios-swift/Messaging/Entities/Conversation.swift
/*
*  Copyright (c) 2011-2019, Zingaya, Inc. All rights reserved.
*/

import Foundation

struct Conversation {
    let uuid: String
    let type: ConversationType
    var title: String
    var participants: [Participant]
    var pictureName: String? // doesnt exist in direct chats
    var description: String? // doesnt exist in direct chats
    var permissions: Permissions? // doesnt exist in direct chats
    var lastUpdated: TimeInterval
    var lastSequence: Int
    let isDirect: Bool
    var isPublic: Bool
    let isUber: Bool
    var latestReadSequence: Int
}

File: client-android-kotlin/messaging/src/main/java/com/voximplant/demos/messaging/entity/Conversation.kt
package com.voximplant.demos.messaging.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.voximplant.demos.messaging.repository.utils.CustomData

@Entity
data class Conversation(
    @PrimaryKey val uuid: String,
    var title: String,
    var participants: MutableList<Long>,
    var lastUpdated: Long,
    var lastSequence: Long,
    val isDirect: Boolean,
    var isPublic: Boolean,
    val isUber: Boolean,
    var customData: CustomData,
)